src/registry/http/blob.rs

```diff
@@ -18,19 +18,26 @@ use tokio::sync::Mutex;
 use tokio_stream::StreamExt;
 use tokio_util::io::ReaderStream;
 
-use super::util::{dump_body_to_string, redirect_uri_fetch};
+use super::util::dump_body_to_string;
 
 #[async_trait::async_trait]
 impl BlobStore for super::HttpRegistry {
     async fn blob_exists(&self, digest: &str) -> Result<bool, Error> {
         let uri = self.repository_uri_from_path(format!("/blobs/{}", digest))?;
 
-        let mut r = redirect_uri_fetch(
-            &self.http_client,
-            |req| req.method(http::Method::HEAD),
-            &uri,
-        )
-        .await?;
+        let mut r = self
+            .http_client
+            .request(
+                &uri,
+                (),
+                |_, c| async {
+                    c.method(http::Method::HEAD)
+                        .body(Body::from(""))
+                        .map_err(|e| e.into())
+                },
+                3,
+            )
+            .await?;
```

Comment:
minor nit: Could we try to refactor this block of sending blank body into a function that takes an http method?

Reply:
I added a `request_simple` method helper and moved these to use that if i understood the request right?